KubeArmor/core/hookHandler.go

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of KubeArmor
+
+package core
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/fsnotify/fsnotify"
+	kl "github.com/kubearmor/KubeArmor/KubeArmor/common"
+	cfg "github.com/kubearmor/KubeArmor/KubeArmor/config"
+	"github.com/kubearmor/KubeArmor/KubeArmor/state"
+	tp "github.com/kubearmor/KubeArmor/KubeArmor/types"
+)
+
+func (dm *KubeArmorDaemon) HandleFile(file string) {
+	f, err := os.Open(file)
+	if err != nil {
+		dm.Logger.Errf("Failed to open file '%s': %v. Ensure the file exists and has appropriate permissions.", file, err.Error())
+	}
+
+	decoder := json.NewDecoder(f)
+	for {
+		var containerData tp.Container
+
+		err = decoder.Decode(&containerData)
+		if err != nil {
+			if err == io.EOF {
+				dm.Logger.Warnf("Reached the end of file '%s'.", file)
+				break
+			}
+			dm.Logger.Errf("Error decoding JSON from file '%s': %v. Verify the file's format and content.", file, err)
+			break
+		}
+		dm.handleContainerCreate(containerData)
+	}
+
+	defer f.Close()
+
+	w, err := fsnotify.NewWatcher()
+	if err != nil {
+		dm.Logger.Errf("Error creating new watcher: (%s)", err.Error())
+	}
+	defer w.Close()
+
+	err = w.Add(file)
+	if err != nil {
+		dm.Logger.Errf("Error adding file to watcher: (%s)", err.Error())
+	}
+
+	for {
+		select {
```

Comment:
We should pass the stop channel and handle that as a case here as well.

See how it's being handled here https://github.com/kubearmor/KubeArmor/blob/a82efe763c3906326cd65a4a01bfba78bb7a960e/KubeArmor/monitor/systemMonitor.go#L683